src/Support/SupportData.php

```diff
@@ -620,11 +658,104 @@ static function ( $url ) {
+				$validation_data = $this->process_raw_post_errors( $amp_validated_post );
+
+				if (
+					! empty( $validation_data['errors'] )
+					&&
+					! empty( $validation_data['error_sources'] )
+					&&
+					! empty( $validation_data['url'] )
+				) {
+					$errors             = array_merge( $errors, $validation_data['errors'] );
+					$error_sources      = array_merge( $error_sources, $validation_data['error_sources'] );
+					$amp_invalid_urls[] = $validation_data['url'];
+				}
+			}
+		} else { // If we have specific URL requested.
+
+			foreach ( $this->urls as $url ) {
+
+				// Check request URL have validated data if not then validate URL.
+				$amp_validated_post_id = post_exists( $url, '', '', AMP_Validated_URL_Post_Type::POST_TYPE_SLUG );
+
+				if ( empty( $amp_validated_post_id ) ) {
+					$validity = AMP_Validation_Manager::validate_url_and_store( $url );
+
+					if ( ! is_array( $validity ) || is_wp_error( $validity ) ) {
+						continue;
+					}
+
+					$amp_validated_post_id = $validity['post_id'];
+				}
+
+				// If validation data is exists for URL then check if those data stalled or not, If stalled then revalidate.
+				$is_stale = ! empty( AMP_Validated_URL_Post_Type::get_post_staleness( $amp_validated_post_id ) );
+				if ( $is_stale ) {
+					AMP_Validation_Manager::validate_url_and_store( $url );
+				}
+
+				$amp_validated_post = get_post( $amp_validated_post_id );
+				$validation_data    = $this->process_raw_post_errors( $amp_validated_post );
+
+				if (
+					! empty( $validation_data['errors'] )
+					&&
+					! empty( $validation_data['error_sources'] )
+					&&
+					! empty( $validation_data['url'] )
+				) {
+					$errors             = array_merge( $errors, $validation_data['errors'] );
+					$error_sources      = array_merge( $error_sources, $validation_data['error_sources'] );
+					$amp_invalid_urls[] = $validation_data['url'];
+				}
+			}
 		}
 
-		$query            = new WP_Query( $query_args );
-		$amp_error_posts  = $query->get_posts();
-		$amp_invalid_urls = [];
+		return [
+			'errors'        => $errors,
+			'error_sources' => $error_sources,
+			'urls'          => $amp_invalid_urls,
+		];
+
+	}
+
+	/**
+	 * Process AMP errors for single AMP validated post.
+	 *
+	 * @param WP_Post $amp_validated_post Instance of AMP Validated Post.
+	 *
+	 * @return array [
+	 *     @type array    $errors             List of error for given instance of Validated post.
+	 *     @type array    $error_sources      List of error sources for given instance of Validated post.
+	 *     @type string   $url                Front-end URL.
+	 * ]
+	 */
+	protected function process_raw_post_errors( WP_Post $amp_validated_post ) {
+
+		if ( AMP_Validated_URL_Post_Type::POST_TYPE_SLUG !== $amp_validated_post->post_type ) {
+			return [];
+		}
+
+		// Empty array for post staleness means post is NOT stale.
+		if ( ! empty( AMP_Validated_URL_Post_Type::get_post_staleness( $amp_validated_post->ID ) ) ) {
```

Comment:
The method takes a `WP_Post` object as well:

```suggestion
		if ( ! empty( AMP_Validated_URL_Post_Type::get_post_staleness( $amp_validated_post ) ) ) {
```